sgkit/io/vcf/vcf_reader.py

```diff
@@ -49,6 +64,163 @@ def get_region_start(region: str) -> int:
     return int(start)
 
 
+def _get_vcf_field_defs(vcf: VCF, category: str) -> Dict[str, Any]:
+    """Get a dictionary of field definitions for a category (e.g. INFO or FORMAT)
+    from the VCF header."""
+    return {
+        h["ID"]: h.info(extra=True)
+        for h in vcf.header_iter()
+        if h["HeaderType"] == category
+    }
+
+
+def _expand_wildcard_fields(vcf: VCF, fields: Sequence[str]) -> Sequence[str]:
+    """Expand 'INFO/*' and 'FORMAT/*' to the full list of fields from the VCF header."""
+    new_fields = []
+    for field in fields:
+        if field == "INFO/*":
+            info_fields = [
+                f"INFO/{key}" for key in _get_vcf_field_defs(vcf, "INFO").keys()
+            ]
+            new_fields.extend(info_fields)
+        elif field == "FORMAT/*":
+            format_fields = set(
+                [f"FORMAT/{key}" for key in _get_vcf_field_defs(vcf, "FORMAT").keys()]
+            )
+            # genotype is handled specially
+            format_fields = format_fields - set(("FORMAT/GT",))
+            new_fields.extend(format_fields)
+        else:
+            new_fields.append(field)
+    return new_fields
+
+
+def _vcf_type_to_numpy_type_and_fill_value(
+    vcf_type: str, category: str, key: str
+) -> Tuple[DType, Any]:
+    """Convert the VCF Type to a NumPy dtype and fill value."""
+    if vcf_type == "Flag":
+        return "bool", False
+    elif vcf_type == "Integer":
+        return "i4", -1
+    elif vcf_type == "Float":
```

Comment:
Worth putting a note in here that BCF stores these values as single precision, so there's no point in using doubles.